app/handlers/retrieveResults.py

```diff
@@ -17,5 +22,38 @@ class RetrieveResultsHandler(BaseHandler):
 
     @apiError error_message The error message to display.
     """
+
     def get(self):
-        self.finish("Retrieve Results")
\ No newline at end of file
+        chunk_size = 2048
+        identifier = self.get_body_argument('identifier')
+        project_path = get_project_path(identifier)
+        file_videos = os.path.join(project_path, 'final_videos')
+        file_images = os.path.join(project_path, 'final_images')
+        file_name = os.path.join(project_path, 'results.zip')
+        dir_len = len(project_path)
+
+        zipf = zipfile.ZipFile(file_name, 'w', zipfile.ZIP_DEFLATED)
+        for root, dirs, files in os.walk(file_videos):
+            for file in files:
+                file_path = os.path.join(root, file)
+                zipf.write(file_path, file_path[dir_len :])
```

Comment:
probably use `os.path.basename` here to get the name for the file rather than `file_path[dir_len:]`